lib/improver/utilities/temporal.py

```diff
@@ -365,3 +365,154 @@ def get_forecast_times(forecast_length, forecast_date=None,
+        Raises:
+            ValueError: If list of times provided falls outside the range
+                        specified by the initial and final times.
+            ValueError: If the interval_in_minutes does not divide the time
+                        range up equally.
+        """
+        time_list = []
+        if self.times is not None:
+            self.times.sort()
+            if self.times[0] < initial_time or self.times[-1] > final_time:
+                raise ValueError(
+                    'List of times falls outside the range given by '
+                    'initial_time and final_time. ')
+
+            for time in self.times:
+                time_list.append(('time', time))
+
+        else:
+            if ((final_time - initial_time).seconds %
+                    (60 * self.interval_in_minutes) != 0):
+                raise ValueError(
+                    'interval_in_minutes provided to time_interpolate does not'
+                    ' divide into the interval equally.')
+
+            time_entry = initial_time
+            while True:
+                time_entry = (time_entry +
+                              timedelta(minutes=self.interval_in_minutes))
+                if time_entry >= final_time:
+                    break
+                time_list.append(('time', time_entry))
+
+        return time_list
+
+    def process(self, cube_t0, cube_t1):
+        """
+        Interpolate data to intermediate times between validity times of
+        cube_t0 and cube_t1.
+
+        Args:
+            cube_t0 (iris.cube.Cube):
+                A diagnostic cube valid at the beginning of the period within
+                which interpolation is to be permitted.
+
+            cube_t1 (iris.cube.Cube):
+                A diagnostic cube valid at the end of the period within which
+                interpolation is to be permitted.
+
+        Returns:
+            interpolated_cubes (iris.cube.CubeList):
+                A list of cubes interpolated to the desired times.
+
+        Raises:
+            CoordinateNotFoundError: The input cubes contain no time
+                                     coordinate.
+            ValueError: The input cubes are ordered such that the initial time
+                        cube has a later validity time than the final cube.
+        """
+        if (not isinstance(cube_t0, iris.cube.Cube) or
+                not isinstance(cube_t1, iris.cube.Cube)):
+            raise TypeError('Inputs to TemporalInterpolation are not of type '
+                            'iris.cube.Cube')
+
+        try:
+            initial_time, = iris_time_to_datetime(cube_t0.coord('time'))
+            final_time, = iris_time_to_datetime(cube_t1.coord('time'))
+        except CoordinateNotFoundError:
+            msg = ('Cube provided to time_interpolate contains no time '
+                   'coordinate.')
+            raise CoordinateNotFoundError(msg)
+        except ValueError:
```

Comment:
This one isn't in your docstring either.

Reply:
✅